### ISSUE TITLE
Alert Provisioning: reference to labels is lost

### ISSUE THREAD

**Issue by aaronstutzer** [grafana/grafana#54850]
**What happened**:
![Screen Shot 2022-09-07 at 15 19 45](https://user-images.githubusercontent.com/23449973/188888435-68da61e1-7cd1-4f6f-a635-27d21feb942d.png)
there should be `$labels` in front of `.name`

**What you expected to happen**:
`{{ $labels.name }}`

When I look at the configmap I see what I see what is expected:
![Screen Shot 2022-09-07 at 15 38 03](https://user-images.githubusercontent.com/23449973/188892423-6b0cba64-e383-40eb-8d0f-5ae47a59a428.png)

**How to reproduce it (as minimally and precisely as possible)**:
Provision an alert with helm-chart
```
grafana:
  alerting:
    platform-argo.yaml: |
      apiVersion: 1
      groups:
        - orgId: 1
          name: Argo
          folder: Alert - Argo
          interval: 60s
          rules:
          - for: 15m
            uid: ARGO003
            labels:
              env: platform
              severity: critical
            annotations:
              summary: "ArgoCD application  {{ `{{ $labels.name }}` }} is in degraded state for more than 15min"
            title: ArgoAppDegradedState
            condition: B
            data:
            - refId: A
              relativeTimeRange:
                from: 600
                to: 0
              datasourceUid: PFA9ECD8827CE8EAA
              model:
                expr: 'argocd_app_info{health_status="Degraded"}'
                hide: false
                intervalMs: 1000
                maxDataPoints: 43200
                refId: A
            - refId: B
              datasourceUid: "-100"
              model:
                conditions:
                - evaluator:
                    params:
                    - 3
                    type: gt
                  operator:
                    type: and
                  query:
                    params:
                    - A
                  reducer:
                    params: []
                    type: last
                  type: query
                datasource:
                  type: __expr__
                  uid: "-100"
                expression: A
                hide: false
                intervalMs: 1000
                maxDataPoints: 43200
                reducer: last
                refId: B
                type: reduce
            - refId: C
              datasourceUid: "-100"
              model:
                conditions:
                - evaluator:
                    params:
                    - 0
                    - 0
                    type: gt
                  operator:
                    type: and
                  query:
                    params: []
                  reducer:
                    params: []
                    type: avg
                  type: query
                datasource:
                  type: __expr__
                  uid: __expr__
                expression: "$B == 1"
                hide: false
                intervalMs: 1000
                maxDataPoints: 43200
                refId: C
                type: math
              intervalSeconds: 60
              no_data_state: OK
              exec_err_state: Alerting
```

**Anything else we need to know?**:
Because of the missing `$labels` the message in Slack looks like:
`ArgoCD application  {{ .name }} is in degraded state for more than 15min`

**Environment**:
- Grafana version: 9.1.3
- Helm-Chart 6.36.3

**Comment by JohnnyQQQQ** [comment]
Thanks for reporting, I have created a PR (#54975) to fix this.

**Comment by JohnnyQQQQ** [comment]
This will be released as part of 9.1.5
